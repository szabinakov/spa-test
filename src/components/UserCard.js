import React from "react";

export default function UserCard({ name, email, avatar }) {
  return (
    <div>
      <p>{name}</p>
      <p>{email}</p>
      <img src={avatar} alt="face of people"></img>
    </div>
  );
}
