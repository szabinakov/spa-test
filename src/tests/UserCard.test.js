import { render } from "@testing-library/react";
import UserCard from "../components/UserCard.js";

describe("UserCard", () => {
  const mockDetails = {
    id: "789",
    name: "MockName",
    email: "MockEmail",
    avatar: "MockURL",
  };
  const mockDeleteBtn = jest.fn();

  it("Matches the snapshot", () => {
    const { asFragment } = render(
      <UserCard
        id={mockDetails.id}
        name={mockDetails.name}
        email={mockDetails.email}
        avatar={mockDetails.avatar}
        deleteUser={mockDeleteBtn}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  const { getByTestId } = render(
    <UserCard
      id={mockDetails.id}
      name={mockDetails.name}
      email={mockDetails.email}
      avatar={mockDetails.avatar}
      deleteUser={mockDeleteBtn}
    />
  );

  expect(getByTestId("userCardContainer")).toHaveClass("userCardContainer");
  expect(getByTestId("tooltiptext")).toHaveClass("tooltiptext");
  expect(getByTestId("iconContainer")).toHaveClass("iconContainer");
  expect(getByTestId("imgContainer")).toHaveClass("imgContainer");
  expect(getByTestId("nameContainer")).toHaveClass("nameContainer");
  expect(getByTestId("emailContainer")).toHaveClass("emailContainer");
  expect(getByTestId("socialDiv")).toHaveClass("socialDiv");
  expect(getByTestId("socialIcons")).toHaveClass("socialIcons");
});
