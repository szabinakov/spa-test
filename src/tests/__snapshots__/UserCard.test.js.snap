// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="userCardContainer"
    data-testid="userCardContainer"
  >
    <button>
      <span
        class="tooltiptext"
        data-testid="tooltiptext"
      >
        Delete User
      </span>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash fa-w-14 iconContainer"
        data-icon="trash"
        data-prefix="fas"
        data-testid="iconContainer"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="imgContainer"
      data-testid="imgContainer"
    >
      <img
        alt="face of people"
        src="MockURL"
      />
    </div>
    <p
      class="nameContainer"
      data-testid="nameContainer"
    >
      MockName
    </p>
    <p
      class="emailContainer"
      data-testid="emailContainer"
    >
      MockEmail
    </p>
    <div
      class="socialDiv"
      data-testid="socialDiv"
    >
      <ul
        class="socialIcons"
        data-testid="socialIcons"
      >
        <li>
          <a
            href="https://www.instagram.com/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-instagram fa-w-14 iconContainer"
              data-icon="instagram"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            href="https://twitter.com/?lang=en-gb"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-twitter fa-w-16 iconContainer"
              data-icon="twitter"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com/feed/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-linkedin fa-w-14 iconContainer"
              data-icon="linkedin"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
        <li>
          <a
            href="https://codepen.io/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-codepen fa-w-16 iconContainer"
              data-icon="codepen"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.285 159.704l-234-156c-7.987-4.915-16.511-4.96-24.571 0l-234 156C3.714 163.703 0 170.847 0 177.989v155.999c0 7.143 3.714 14.286 9.715 18.286l234 156.022c7.987 4.915 16.511 4.96 24.571 0l234-156.022c6-3.999 9.715-11.143 9.715-18.286V177.989c-.001-7.142-3.715-14.286-9.716-18.285zM278 63.131l172.286 114.858-76.857 51.429L278 165.703V63.131zm-44 0v102.572l-95.429 63.715-76.857-51.429L234 63.131zM44 219.132l55.143 36.857L44 292.846v-73.714zm190 229.715L61.714 333.989l76.857-51.429L234 346.275v102.572zm22-140.858l-77.715-52 77.715-52 77.715 52-77.715 52zm22 140.858V346.275l95.429-63.715 76.857 51.429L278 448.847zm190-156.001l-55.143-36.857L468 219.132v73.714z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
